test/rulesets/SparseArrays/sparsematrix.jl

```diff
@@ -18,6 +18,51 @@ end
     test_rrule(SparseVector{Float32}, Float32.(v), rtol=1e-4)
 end
 
+# copied over from test/rulesets/LinearAlgebra/structured
+@testset "spdiagm" begin
+    @testset "without size" begin
+        M, N = 7, 9
+        s = (8, 8)
+        a, ā = randn(M), randn(M)
+        b, b̄ = randn(M), randn(M)
+        c, c̄ = randn(M - 1), randn(M - 1)
```

Comment:
```suggestion
        a = randn(M)
        b = randn(M)
        c = randn(M - 1)
```
The barred variables are not used.